Tools/Scripts/webkitdirs.pm

```diff
@@ -965,8 +965,10 @@ sub determineXcodeSDKPlatformName {
     if (checkForArgumentAndRemoveFromARGVGettingValue("--sdk", \$sdk)) {
         $xcodeSDK = lc $sdk;
         $xcodeSDKPlatformName ||= $sdk;
-        $xcodeSDKPlatformName =~ s/\.internal$//;
+        $xcodeSDKPlatformName =~ s/(\d+\.[\d\.]+)?(\.internal)?$//;
         die "Couldn't determine platform name from Xcode SDK" unless isValidXcodeSDKPlatformName($xcodeSDKPlatformName);
+        `xcrun --sdk $xcodeSDK --show-sdk-path 2>&1`;
+        die "Unable to find requested sdk '" . $xcodeSDK . "' requested" if exitStatus($?);
```

Comment:
Talked privately but: I've tried to avoid making calls to `xcrun` when executing `XcodeOptions()`, since xcrun can be a little slow (~2 sec) when uncached. So, I'd suggest to not unconditionally run this, especially since most people pass an unversioned SDK name.